src/SendGrid/SendGridClient.cs

```diff
@@ -51,7 +53,22 @@ public SendGridClient(HttpClient httpClient, string apiKey, string host = null,
         /// <param name="urlPath">Path to endpoint (e.g. /path/to/endpoint).</param>
         /// <returns>Interface to the Twilio SendGrid REST API.</returns>
         public SendGridClient(string apiKey, string host = null, Dictionary<string, string> requestHeaders = null, string version = null, string urlPath = null)
-            : base(buildOptions(apiKey, host, requestHeaders, version, urlPath))
+            : base(buildOptions(apiKey, host, requestHeaders, version, urlPath, false))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SendGridClient"/> class.
+        /// </summary>
+        /// <param name="apiKey">Your Twilio SendGrid API key.</param>
+        /// <param name="host">Base url (e.g. https://api.sendgrid.com).</param>
+        /// <param name="requestHeaders">A dictionary of request headers.</param>
+        /// <param name="version">API version, override AddVersion to customize.</param>
+        /// <param name="urlPath">Path to endpoint (e.g. /path/to/endpoint).</param>
+        /// <param name="httpErrorAsException">Indicates whether HTTP error responses should be raised as exceptions. Default is false.</param>
+        /// <returns>Interface to the Twilio SendGrid REST API.</returns>
+        public SendGridClient(string apiKey, bool httpErrorAsException = false, string host = null, Dictionary<string, string> requestHeaders = null, string version = null, string urlPath = null)
```

Comment:
Not liking `httpErrorAsException` now being the second arg since it's last everywhere else. Was this the conflict you saw and what was the conflict with?

If it can't be the last arg, recommend just dropping this new constructor since we already have constructors below which take in a `SendGridClientOptions` instance which can be used to achieve the necessary effect.

Reply:
This issue was that a test was using the existing constructor with just the first arguments, and then the compiler said the call was ambiguous because those first arguments match both constructors.

https://github.com/sendgrid/sendgrid-csharp/actions/runs/2756351061

So basically having having an overload with an extra optional parameter at the end means that the first constructor effectively cannot be used, because it would be ambiguous  with the next constructor.

But you are right, the SendGridClientOptions overload gives the same features as this constructor. I guess the constructors could be consolidated for a major breaking release at some point to avoid confusion.

I will remove the new overload.